tests/mir-opt/jump_threading.rs

```diff
@@ -535,11 +535,17 @@ pub fn bitwise_not() -> i32 {
     // CHECK-LABEL: fn bitwise_not(
 
     // Test for #131195, which was optimizing `!a == b` into `a != b`.
-    let mut a: i32 = 0;
```

Comment:
Just removed the mutation to simplify the test since it wasn't necessary in the original demo